frontend/__tests__/src/components/Common/NotificationButton/NotificationButton.test.tsx

```diff
@@ -0,0 +1,101 @@
+import { cleanup, fireEvent, render, renderHook, waitFor } from '@testing-library/react'
+import { NotificationButton } from '@src/components/Common/NotificationButton/NotificationButton'
+import React from 'react'
+import { useNotificationLayoutEffect } from '@src/hooks/useNotificationLayoutEffect'
+import { act } from 'react-dom/test-utils'
+import { HEADER_NOTIFICATION_MESSAGE } from '@src/constants'
+
+describe('NotificationButton', () => {
+  afterEach(cleanup)
+  const closeNotificationProps = { open: false, title: 'NotificationPopper' }
+  const openNotificationProps = { open: true, title: 'NotificationPopper' }
+  const { result } = renderHook(() => useNotificationLayoutEffect())
+
+  it('should show NotificationIcon when render NotificationButton component', () => {
+    const { getByTestId } = render(<NotificationButton {...result.current} />)
+
+    expect(getByTestId('NotificationIcon')).toBeInTheDocument()
+  })
+
+  it('should hide NotificationIcon when notificationProps is undefined', () => {
+    act(() => {
+      result.current.notificationProps = undefined
+    })
+    const { queryByTestId } = render(<NotificationButton {...result.current} />)
+
+    expect(queryByTestId('NotificationIcon')).not.toBeInTheDocument()
+  })
+
+  it('should show NotificationPopper when clicking the component given the "open" value is true', async () => {
+    act(() => {
+      result.current.notificationProps = openNotificationProps
+    })
+    const { getByTestId, getByText } = render(<NotificationButton {...result.current} />)
+
+    fireEvent.click(getByTestId('NotificationIcon'))
+    expect(getByText('NotificationPopper')).toBeInTheDocument()
+  })
+
+  it('should hide NotificationPopper when clicking the component  given the "open" value is false', async () => {
+    act(() => {
+      result.current.notificationProps = closeNotificationProps
+    })
+    const { getByTestId, queryByText } = render(<NotificationButton {...result.current} />)
+
+    fireEvent.click(getByTestId('NotificationIcon'))
```

Comment:
1. same issue.
2. Extract `NotificationIcon` as a fixture or a global variable in this file